### PR TITLE
Docs: Formatting tweaks and fixes

### PR DESCRIPTION
**Pull request by WardBrian**
This tries to apply a more consistent style to the docs:

- Use more of the [Sphinx directives](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html) for things like files
- Better cross-linking of functions to each other
- I changed the highlight color for `code items` to a shade of green rather than the default pink
- I fixed up some paragraphs which were outdated or missing information

### CODE REVIEW

**Comment by WardBrian** [comment]
Preview: https://brianward.dev/bridgestan/latest/index.html

File: docs/_static/css/custom.css
Comment (roualdes) [inline docs/_static/css/custom.css:1]: I much prefer the green in dark mode and slightly prefer the pink in light mode. Ha.  I guess that's still a net win.  Did you try light mode with this?

File: docs/internals/testing.rst [hunk 1]
```diff
@@ -68,3 +68,13 @@ R tests are written using `testthat <https://testthat.r-lib.org/>`__.
     Rscript -e "devtools::test()"
 
 The R unit tests are much more basic than the Python or Julia tests.
+
+Rust
+_____
+
+The R tests can be run with :command:`cargo`
```
Comment (roualdes) [inline docs/internals/testing.rst:75]:
```suggestion
The Rust tests can be run with :command:`cargo`
```

**Review by roualdes — CHANGES_REQUESTED** [review]
My only real question is for you to double check the pink to green inline code color change.  Otherwise, the added links and nuance between inline code, file paths, make variables, and commands are great.

**Review by roualdes — APPROVED** [review]
I like that green much better.  Thanks for the nit-picky color preferences.

Feel free to merge once checks pass.

**Pull request merged**